packages/cli/src/ui/App.tsx

```diff
@@ -559,7 +565,16 @@ const App = ({ config, settings, startupWarnings = [], version }: AppProps) => {
     [handleSlashCommand],
   );
 
+  const handleCtrlCWithEmptyBuffer = useCallback(() => {
+    handleExit(ctrlCPressedOnce, setCtrlCPressedOnce, ctrlCTimerRef);
+  }, [ctrlCPressedOnce, handleExit]);
+
   useInput((input: string, key: InkKeyType) => {
+    // if ESC key, don't handle it in App, let InputPrompt handle it
```

Comment:
this comment doesn't seem right. I would remove this block as it is a no-op. whether we return or not has no baring on how InputPrompt does input handling.